### ISSUE TITLE
Progressively get rid of MD5

### ISSUE THREAD

**Issue by bloatware** [textpattern/textpattern#1895]
### Is your feature request related to a problem?

Potentially.

### What is the feature?

We currently use MD5 for all sort of hashes, both cryptographic and not. But

- MD5 is not very reliable as crypto hash and could be replaced by Tiger-128, which is slightly faster too.
- PHP8 implements XXH algo for non-crypto hashes, which is *way* faster.

The only drawback of replacing MD5 seems to be that we use it also in some db queries, but this is not essential.

Thoughts?

**Comment by Bloke** [comment]
I'm all for migrating. We do need to harden the login session cookies anyway because it's a bit old hat. Since the move to a better algorithm will invalidate login cookies, it makes sense to do both at the same time to minimise disruption.

**Comment by bloatware** [comment]
Yep, 4.9 is probably a good moment to do it, since we tighten the security here and there. The public cookie is especially weak. Tokens are concerned too.

**Comment by petecooper** [comment]
Given the changes made in https://github.com/textpattern/textpattern/commit/7a38261abaf442fb8a8271994817210d7cbd60df - is this viable for 4.9 series? I appreciate there's a variety of places where `md5()` is used (articles, cookies, tokens, mail, etc), and we'll need to factor in subsequent pophelp changes, too.
